src/Jackett.Common/Utils/CookieUtil.cs

```diff
@@ -0,0 +1,39 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Jackett.Common.Utils
+{
+    public static class CookieUtil
+    {
+        // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+        private static readonly Regex _CookieRegex = new Regex(@"([^\(\)<>@,;:\\""/\[\]\?=\{\}\s]+)=([^,;\\""\s]+)");
+        private static readonly char[] InvalidKeyChars = {'(', ')', '<', '>', '@', ',', ';', ':', '\\', '"', '/', '[', ']', '?', '=', '{', '}', ' ', '\t', '\n'};
+        private static readonly char[] InvalidValueChars = {'"', ',', ';', '\\', ' ', '\t', '\n'};
+
+        public static Dictionary<string, string> CookieHeaderToDictionary(string cookieHeader)
+        {
+            var cookieDictionary = new Dictionary<string, string>();
+            if (cookieHeader == null)
+                return cookieDictionary;
+            var matches = _CookieRegex.Match(cookieHeader);
+            while (matches.Success)
+            {
+                if (matches.Groups.Count > 2)
+                    cookieDictionary[matches.Groups[1].Value] = matches.Groups[2].Value;
+                matches = matches.NextMatch();
+            }
+            return cookieDictionary;
+        }
+
+        public static string CookieDictionaryToHeader(Dictionary<string, string> cookieDictionary)
+        {
+            if (cookieDictionary == null)
+                return "";
+            return string.Join(
+                "; ", cookieDictionary
+                      .Where(kv => kv.Key.IndexOfAny(InvalidKeyChars) < 0 && kv.Value.IndexOfAny(InvalidValueChars) < 0 )
```

Comment:
Based on your given test cases below, maybe completely skipping malformed keys/values might not be the best route? I feel like maybe web encoding or truncating would potentially be better. Just looking through your test cases, I would have thought `{"__cf_bm", "34234 234"}` would become either `"__cf_bm=34234"` or `"__cf_bm=34234+234"` since these dictionaries would be built either by your conversion code, or by devs themselves, it feels like `Dictionary->string` would be more on purpose with choice of key/value.

Reply:
Well, I choose a silent approach because the previous code never fails. Changing the behavior can cause problems if some indexer is wrong.
I'm going to change it to raise an exception instead, the user will be responsible for encoding the value properly. Most of the cookies we use come from webservers and they are right.

Reply:
Changed, I tested all public trackers and is fine throwing the exception.

Reply:
That works